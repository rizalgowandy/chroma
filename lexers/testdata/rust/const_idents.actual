const IM_A_CONST: () = ();

fn main() {
    dbg!(IM_A_CONST);
}

// None of this should be considered const
fn foo<T0, T1, U>(_: (T0, T1, U)) {
    let _: std::collections::BTreeSet<()> = todo!();
}
