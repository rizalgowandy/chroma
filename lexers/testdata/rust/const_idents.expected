[
  {"type":"Keyword","value":"const"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameConstant","value":"IM_A_CONST"},
  {"type":"Text","value":": "},
  {"type":"Punctuation","value":"()"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"();"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"fn"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"main"},
  {"type":"Punctuation","value":"()"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"NameFunctionMagic","value":"dbg!"},
  {"type":"Punctuation","value":"("},
  {"type":"NameConstant","value":"IM_A_CONST"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"// None of this should be considered const\n"},
  {"type":"Keyword","value":"fn"},
  {"type":"Text","value":" "},
  {"type":"NameFunction","value":"foo"},
  {"type":"Operator","value":"\u003c"},
  {"type":"Name","value":"T0"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"T1"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"U"},
  {"type":"Operator","value":"\u003e"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"_"},
  {"type":"Text","value":": "},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"T0"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"T1"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"U"},
  {"type":"Punctuation","value":"))"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"KeywordDeclaration","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"_"},
  {"type":"Text","value":": "},
  {"type":"NameClass","value":"std"},
  {"type":"Text","value":"::"},
  {"type":"Name","value":"collections"},
  {"type":"Text","value":"::"},
  {"type":"Name","value":"BTreeSet"},
  {"type":"Operator","value":"\u003c"},
  {"type":"Punctuation","value":"()"},
  {"type":"Operator","value":"\u003e"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameFunctionMagic","value":"todo!"},
  {"type":"Punctuation","value":"();"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
